p2p/src/main/kotlin/maru/p2p/MaruPeerManager.kt

```diff
@@ -35,7 +40,10 @@ import tech.pegasys.teku.networking.p2p.peer.Peer
 class MaruPeerManager(
```

Comment:
I feel like it's a poor choice to pile more responsibilities onto `MaruPeerManager`. The code here is getting more and more spaghetti and `MaruPeerManager` is becoming a god object. While there's not that much code, given the asynchronicity and all the mutable state, the complexity is quite big.

I suggest to refactor it in such a way so that pieces that are responsible for the specific parts of the business logic (status update, forkId validation, dropping not useful peers) are all in separate classes, that _interact_ with `MaruPeerManager`, rather than be a part of it.

It's becoming very hard to review and it already was hard for me to maintain when I was trying to fix a flacky test.